addons/ofxiOS/src/gl/ES3Renderer.m

```diff
@@ -0,0 +1,189 @@
+
+
+#import "ES3Renderer.h"
+
+@implementation ES3Renderer
+
+// Create an OpenGL ES 3.0 context
+- (id)init {
+	return [self initWithDepth:false andAA:false andFSAASamples:0 andRetina:false];
+}
+
+- (id)initWithDepth:(bool)depth andAA:(bool)fsaa andFSAASamples:(int)samples andRetina:(bool)retina {
+	if((self = [super init])) {
+		depthEnabled = depth;
+		fsaaEnabled = fsaa;
+		fsaaSamples = samples;
+		retinaEnabled = retina;
+		bResize = false;
+		
+		context = [[EAGLContext alloc] initWithAPI:kEAGLRenderingAPIOpenGLES3];
+		NSLog(@"Creating OpenGL ES3 Renderer");
+		
+		if(!context || ![EAGLContext setCurrentContext:context]) {
+			NSLog(@"OpenGL ES3 failed");
+			[self release];
+			return nil;
+		}
+		
+		const GLubyte * extensions = glGetString(GL_EXTENSIONS);
+		if(extensions != NULL && fsaaEnabled) {
+			if(strstr((const char*)extensions, "GL_APPLE_framebuffer_multisample")) {
+				fsaaEnabled = true;
+			} else {
+				fsaaEnabled = false;
+			}
+		} else {
+			fsaaEnabled = false;
+		}
+	}
+	
+	
+	return self;
+}
+
+-(EAGLContext*) context {
+	return context;
+}
+
+- (void)startRender {
+	[EAGLContext setCurrentContext:context];
+}
+
+- (void)finishRender {
+	if(fsaaEnabled) {
+		if(depthEnabled) {
+			GLenum attachments[] = {GL_COLOR_ATTACHMENT0, GL_DEPTH_ATTACHMENT};
+			glInvalidateFramebuffer(GL_READ_FRAMEBUFFER_APPLE, 2, attachments);
+		} else {
+			GLenum attachments[] = {GL_COLOR_ATTACHMENT0};
+			glInvalidateFramebuffer(GL_READ_FRAMEBUFFER_APPLE, 1, attachments);
+		}
+		
+		glBindFramebuffer(GL_READ_FRAMEBUFFER_APPLE, fsaaFrameBuffer);
+		glBindFramebuffer(GL_DRAW_FRAMEBUFFER_APPLE, defaultFramebuffer);
+		glBlitFramebuffer(0, 0, backingWidth, backingHeight, 0, 0, backingWidth, backingHeight, GL_COLOR_BUFFER_BIT, GL_NEAREST);
```

Comment:
This is new anti-aliasing code for ES3 renderer, I'm still not sure if its working. Need to verify it is.